core/src/main/java/hudson/model/Run.java

```diff
@@ -660,32 +671,43 @@ public final long getStartTimeInMillis() {
     }
 
     @Exported
+    @CheckForNull
     public String getDescription() {
         return description;
     }
 
 
     /**
      * Returns the length-limited description.
+     * The method tries to take the HTML tags into account, but it is a best-effort attempt.
+     * Also, the method will not work properly if a non-HTML {@link hudson.markup.MarkupFormatter} is used.
      * @return The length-limited description.
-     * @deprecated truncated description uses arbitrary and unconfigurable limit of 100 symbols
+     * @deprecated truncated description based on the {@link #TRUNCATED_DESCRIPTION_LIMIT} setting.
```

Comment:
The deprecation doesn't really make sense anymore? At least for a while?

Reply:
I do not want to revert it for now, will probably act on that in a follow-up PR